### PR TITLE
Improve copyright rule

### LINKED ISSUE #1166: Bug in rule “headers.copyright”:  ignores the license used
Checker gives this error

> Missing link 'https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer' with text 'liability'.
The copyright must use the following markup (fill in with the appropriate year, years, or year range).

> Copyright © 2021 W3C ® ( MIT , ERCIM , Keio, Beihang). W3C liability, trademark and document use rules apply.
> Include this source code:

```html
<p class="copyright"><a href="https://www.w3.org/Consortium/Legal/ipr-notice#Copyright">
Copyright</a> © 2021 <a href="https://www.w3.org/">
<abbr title="World Wide Web Consortium">W3C</abbr></a><sup>®</sup>
 (<a href="https://www.csail.mit.edu/"><abbr title="Massachusetts 
Institute of Technology">MIT</abbr></a>, <a href="https://www.ercim.eu/">
<abbr title="European Research Consortium for Informatics and 
Mathematics">ERCIM</abbr></a>, <a href="https://www.keio.ac.jp/">Keio</a>, 
<a href="https://ev.buaa.edu.cn/">Beihang</a>). 
W3C <a href="https://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer">liability</a>, 
<a href="https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks">trademark</a> and 
<a href="https://www.w3.org/Consortium/Legal/copyright-documents">document use</a> rules apply.</p>
```

While the (Bikeshed-generated)  document actually contains

```html
<p class="copyright" data-fill-with="copyright"><a href="https://www.w3.org/Consortium/Legal/ipr-notice#Copyright">
Copyright</a> © 2021 <a href="https://www.w3.org/">
<abbr title="World Wide Web Consortium">W3C</abbr></a><sup>®</sup>
 (<a href="https://www.csail.mit.edu/"><abbr title="Massachusetts 
Institute of Technology">MIT</abbr></a>, <a href="https://www.ercim.eu/">
<abbr title="European Research Consortium for Informatics and 
Mathematics">ERCIM</abbr></a>, <a href="https://www.keio.ac.jp/">Keio</a>, 
<a href="https://ev.buaa.edu.cn/">Beihang</a>). 
W3C <a href="http://www.w3.org/Consortium/Legal/ipr-notice#Legal_Disclaimer">liability</a>, 
<a href="https://www.w3.org/Consortium/Legal/ipr-notice#W3C_Trademarks">trademark</a> and 
<a href="https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document" 
rel="license">permissive document license</a> rules apply. </p>
```

The difference which the checker sees, is that the disclaimer link uses https instead of http, which is easily fixed.

Raising this issue to report that, had I done as the checker suggested and blindly copy-pasted the *must be used* markup, it would have changed the license from **W3C Software and Document** to **W3C Document**.

Found [while checking `https://www.w3.org/TR/2021/CR-css-scroll-snap-1-20210311/`](https://www.w3.org/pubrules/?url=https%3A%2F%2Fwww.w3.org%2FTR%2F2021%2FCR-css-scroll-snap-1-20210311%2F&profile=CR&validation=simple-validation&noRecTrack=false&informativeOnly=false&echidnaReady=false&patentPolicy=pp2020).

### PR DESCRIPTION
**Pull request by jennyliang220**
Making sure the copyright mentioned in the document is consistant with the Charter.
resolves #1166 

Also split the document for copyright:
6081626774061_ pic_hd

### CODE REVIEW

**Comment by jennyliang220** [comment]
Failed lint because this PR is using the rule changed in https://github.com/w3c/specberus/pull/1326. After merging #1326, the check would be good.

File: lib/l10n-en_GB.js [hunk 1]
```diff
@@ -120,11 +120,17 @@ exports.messages = {
...
-    'headers.copyright.kitten-friendly': 'Using CC-BY kitten-friendly license.',
-    'headers.copyright.link-text':
-        "Missing link '${href}' with text '${text}'.",
+        'Copyright string does not match requirements. The string should match <blockquote class="boilerplate">${rex}</blockquote>',
+    'headers.copyright.no-link': "Missing link with text '${text}'.",
+    'headers.copyright.href-not-match':
+        "The link for text '${text}' is expected to be '${expected}', but found '${hrefInDoc}' in document.",
     'headers.copyright.not-found': 'Missing copyright paragraph.',
+    'headers.copyright.no-data-from-API':
+        "Could not get the document license from the API. Please make sure this group has a current charter, and it's mentioning at least one document license.",
```
Comment (deniak) [inline lib/l10n-en_GB.js:129]:
```suggestion
        "Could not get the document license from the API. Please make sure this group has a current charter associated with at least one document license.",
```

File: lib/l10n-en_GB.js [hunk 2]
```diff
@@ -120,11 +120,17 @@ exports.messages = {
...
+    'headers.copyright.no-license-found-joint':
+        "This document is published by multiple groups, but these groups doesn't agree on the same document license. Please ask the team contact for help.",
```
Comment (deniak) [inline lib/l10n-en_GB.js:131]:
```suggestion
        "This document is published by multiple groups, but these groups don't use the same document license. Please ask the team contact for help.",
```

File: lib/rules/headers/copyright.js [hunk 3]
```diff
@@ -1,88 +1,148 @@
+/**
+ * @file check if the document uses right copyright.
+ * @description for this rule.
+ * 1. Get type of license(s) the group is using from the API.
+ * 2. There're 2 types of license(s): copyright-documents and copyright-software. A group would choose either one or both in the Charter. If they choose both, they can mention either one in the document.
+ * 3. For joint-publications, the license should be adopted by both groups.
+ * 4. For "copyright-software", the url is https://www.w3.org/Consortium/Legal/copyright-software, the dated url is https://www.w3.org/Consortium/Legal/2015/copyright-software-and-document, they are both allowed. The name in the API is "W3C Software and Document License", but the document would use text "permissive document license".
+ * 5.For "copyright-documents", the url is https://www.w3.org/Consortium/Legal/copyright-documents. The name in the API is "W3C Document License", but the document would use text "document use".
```
Comment (deniak) [inline lib/rules/headers/copyright.js:8]:
```suggestion
 * 5. For "copyright-documents", the url is https://www.w3.org/Consortium/Legal/copyright-documents. The name in the API is "W3C Document License", but the document would use text "document use".
```

File: lib/rules/headers/copyright.js [hunk 4]
```diff
@@ -1,88 +1,148 @@
...
+            } else {
+                // Find union of multiple groups' license. Delete license in base if not adopted by other groups.
+                allowedLicenses = allowedLicenses.filter(allowedLicense =>
+                    // this allowedLicense is also adopted by group
+                    charterData['doc-licenses'].some(
+                        licenseObj => allowedLicense === licenseObj.uri
+                    )
+                );
+            }
+        });
```
Comment (deniak) [inline lib/rules/headers/copyright.js:67]:
```suggestion
        const licenses = [...charterData.map((c => c['doc-licenses'].map(l => l.uri)))]; // creates an array of license URIs per group -> array of arrays
        const allowedLicenses = licenses.reduce((a, b) => a.filter(c => b.includes(c))); // find the intersection of all arrays
```

File: lib/rules/headers/copyright.js [hunk 5]
```diff
@@ -1,88 +1,148 @@
...
+
+        if (!allowedLicenses.length) {
+            if (isJointPublication) {
+                sr.error(self, 'no-license-found-joint');
+                return done();
+            }
+            sr.error(self, 'no-license-found');
```
Comment (deniak) [inline lib/rules/headers/copyright.js:74]:
```suggestion
            sr.error(self, (isJoinPublication) ? 'no-license-found-joint' : 'no-license-found');
```

File: lib/rules/headers/copyright.js [hunk 6]
```diff
@@ -1,88 +1,148 @@
...
-            doMeanwhile() {
-                w3cOnlyStatement = '(, All Rights Reserved)?';
-            },
-            doAfter() {
-                w3cOnlyStatement = '';
-            },
-        });
+        const regResult = text.match(rex);
+        let licenseInDoc;
+        if (!regResult) sr.error(self, 'no-match', { rex });
```
Comment (deniak) [inline lib/rules/headers/copyright.js:108]: Shouldn't we stop the rule if there's no match?

**Review by deniak — CHANGES_REQUESTED** [review]

File: lib/rules/headers/copyright.js (hunk 5 above)
Comment (deniak) [reply lib/rules/headers/copyright.js:63]: it's actually breaking the l10n test

**Review by deniak — APPROVED** [review]

**Pull request merged**